packages/core/src/alert/alert-actions.element.scss

```diff
@@ -1,8 +1,66 @@
+@import './../styles/tokens/generated/index';
 @import './../styles/mixins/utils';
 
 :host {
-  flex: 0 0 auto;
+  --action-text-color: #{$cds-token-color-neutral-700};
+  --action-hover-text-color: #{$cds-token-color-neutral-1000};
+  --action-size: auto;
+  --font-size: #{$cds-token-typography-font-size-3};
+
   white-space: nowrap;
 
   // TODO: add style override for dropdown component inside alert-actions after cds-dropdown is implemented
 }
+
+:host([type='light']) {
+  display: none !important;
+}
+
+::slotted(cds-button.alert-btn) {
+  --color: var(--action-text-color, #{$cds-token-color-neutral-700-static});
+  --border-color: var(--action-text-color, #{$cds-token-color-neutral-700-static});
+  --background: none;
+  --padding: none;
+  --box-shadow-color: transparent;
+  --text-transform: none;
+  --text-decoration: underline;
+  --border-width: 0;
+  --height: var(--action-size);
+  --min-width: initial;
+
+  display: inline-block;
+}
+
+::slotted(cds-button.alert-btn:hover) {
+  --color: var(--action-hover-text-color, #{$cds-token-color-neutral-1000-static});
+  --border-color: var(--action-hover-text-color, #{$cds-token-color-neutral-1000-static});
+}
+
+:host([type='default']) {
+  --action-size: calc(#{$cds-token-space-size-9} - #{$cds-token-space-size-3});
+  padding-top: #{$cds-token-space-size-2};
```

Comment:
should the `padding-top` be defined by its container rather than in the component?

Reply:
The container is outside of the slot while the component here is projected into the slot for alert actions.

When I put padding on the container, it was like it then ignored the layouts in the slot (pushing everything down much farther than we wanted). It's also worth noting that this particular fix was related to issues I was seeing in Safari. It was a late entry because things looked fine everywhere else but Safari was not handling padding on the container and then negative margin on a layout projected inside the slot gracefully.

Safari was like "oh.. you must not want any negative margins then!" on the slot content. 😬

Reply:
For future reference... we should check how things are laying out in Safari. It's no longer a to-be-trusted platform. Things honestly look great and consistent across firefox, (new) edge, and chrome. Safari is quickly becoming a problem...

Reply:
Thank you for the explanation. It's too bad that Safari has become the new "outlier browser."  😞  I'll be looking out for that in my work too, to be extra vigilant in checking that it looks as expected in Safari.